packages/jest-resolve/src/resolver.ts

```diff
@@ -309,12 +240,143 @@ export default class Resolver {
     this._modulePathCache.set(from, paths);
     return paths;
   }
+}
 
-  getModuleID(
+export class ResolverAsync extends BaseResolver {
```

Comment:
I think I'd prefer a single exported `class` with both sync and async methods on it rather than 2 separate classes